File: src-react/src/lib/api/prompts.ts
import type { AgentName, Prompt } from "@/types";
import { invokeCommand } from "@/lib/api/tauri";

export interface SetPromptTargetInput {
  promptId: string;
  agent: AgentName;
  enabled: boolean;
}

export interface MovePromptSourceInput {
  promptId: string;
  agent: AgentName;
}

export const promptsApi = {
  list(): Promise<Prompt[]> {
    return invokeCommand<Prompt[]>("list_prompts");
  },

  scan(): Promise<Prompt[]> {
    return invokeCommand<Prompt[]>("scan_prompts");
  },

  setTarget(input: SetPromptTargetInput): Promise<Prompt> {
    return invokeCommand<Prompt>("set_prompt_target", { input });
  },

  moveSource(input: MovePromptSourceInput): Promise<Prompt> {
    return invokeCommand<Prompt>("move_prompt_source", { input });
  },

  openSource(id: string): Promise<void> {
    return invokeCommand<void>("open_prompt_source", { id });
  },

  revealPath(id: string): Promise<void> {
    return invokeCommand<void>("reveal_prompt_path", { id });
  },
};

File: src-tauri/src/commands/prompts.rs
use tauri::State;

use nexus_core::{
    error::AppResult,
    services::prompts::{Prompt, SetPromptTargetInput},
};

use crate::store::AppState;

#[tauri::command]
pub fn list_prompts(state: State<'_, AppState>) -> AppResult<Vec<Prompt>> {
    state.prompts.list_prompts()
}

#[tauri::command]
pub fn scan_prompts(state: State<'_, AppState>) -> AppResult<Vec<Prompt>> {
    state.prompts.scan_prompts()
}

#[tauri::command]
pub fn set_prompt_target(
    state: State<'_, AppState>,
    input: SetPromptTargetInput,
) -> AppResult<Prompt> {
    state.prompts.set_prompt_target(input)
}

#[tauri::command]
pub fn open_prompt_source(state: State<'_, AppState>, id: String) -> AppResult<()> {
    state.prompts.open_prompt_source(id)
}

#[tauri::command]
pub fn reveal_prompt_path(state: State<'_, AppState>, id: String) -> AppResult<()> {
    state.prompts.reveal_prompt_path(id)
}
